lib/matching/nearest_neighbour.cpp

```diff
@@ -6,13 +6,40 @@
  */
 
 #include "nearest_neighbour.h"
+#include "Eigen/Core"
+#include <limits.h>
 
 namespace MouseTrack {
 
-std::vector<long> NearestNeighbour::operator()(const std::vector<std::shared_ptr<const ClusterDescriptor>>& descriptors, const std::vector<ClusterChain>& chains) {
-	return std::vector<long>{};
+std::vector<long> NearestNeighbour::operator()(
+    const std::vector<std::shared_ptr<const ClusterDescriptor>> &descriptors,
+    const std::vector<ClusterChain> &chains) {
+  // Prepare an empty vector that will store the index of each chain
+  std::vector<long> matchIndex;
+  for (size_t j = 0; j < descriptors.size(); j++) {
+    // Initialize a maximum distance
+    double nearest = std::numeric_limits<double>::max();
+    for (size_t i = 0; i < chains.size(); i++) {
+      // In this case we are only really concerned with the last cluster. Get
+      // the frame.
+      FrameNumber lastFrameNumber = chains[i].descriptors().rbegin()->first;
```

Comment:
What if the chain has no descriptors?

1. Either check for this case in the algorithm.
2. Or add a note in the comment of the public interface, that no empty chains are allowed.

I tend to go with 2.: It's more efficient and an easy thing to fulfill.
Arguments for 1: It's more robust and more general, it behaves in an intuitive way.